dirsrvtests/tests/suites/schema/schema_replication_origin_test.py

```diff
@@ -0,0 +1,234 @@
+        topo_roles = {ReplicaRole.SUPPLIER: 1, ReplicaRole.CONSUMER: 1}
+    elif request.param == "m1h1c1":
+        topo_roles = {ReplicaRole.SUPPLIER: 1, ReplicaRole.HUB: 1, ReplicaRole.CONSUMER: 1}
+
+    topology = create_topology(topo_roles, request=request)
+
+    topology.logcap = LogCapture()
+    return {
+        'topology': topology, 
+        'type': request.param
+    }
+
+
+@pytest.fixture(scope="function")
+def schema_replication_init(topology):
+    """Initialize the test environment """
+
+    supplier = topology['topology'].ms["supplier1"]
+    supplier.errorlog_file = open(supplier.errlog, "r")
+    users = UserAccounts(supplier, DEFAULT_SUFFIX)
+    user = users.create_test_user()
+    user.replace('telephonenumber', '0')
+    
+    return user
+
+
+@pytest.mark.parametrize("xorigin", ['user defined', 'custom xorigin'])
+def test_schema_xorigin_repl(topology, schema_replication_init, xorigin):
+    """Check consumer schema is a superset (one extra OC) of supplier schema, then
+    schema is pushed and there is a message in the error log
+
+    :id: 2b29823b-3e83-4b25-954a-8a081dbc15ee
+    :setup: Supplier and consumer topology, with one user entry;
+            Supplier, hub and consumer topology, with one user entry
+    :steps:
+        1. Push the schema from the supplier to the consumer (an error should not be generated)
+        2. Update the schema of the consumer, so it will be a superset of the supplier's schema
+        3. Update the schema of the supplier to make its nsSchemaCSN larger than the consumer's
+        4. Push the schema from the supplier to the consumer (an error should be generated)
+        5. Check if the supplier learns the missing definition
+        6. Check the error logs for any issues
+        7. Check the startup and final state of the schema replication process
+    :expectedresults:
+        1. The supplier's schema update should be successful
+        2. The consumer's schema update should be successful
+        3. The supplier's schema update should be successful
+        4. The schema push operation should be successful
+        5. The supplier should successfully learn the missing definition
+        6. There should be no error messages in the logs
+        7. The startup and final state of the schema replication process should be as expected
+    """
+
+    user = schema_replication_init
+    hub = None
+    supplier = topology['topology'].ms["supplier1"]
+    consumer = topology['topology'].cs["consumer1"]
+    if topology['type'] == "m1h1c1":
+        hub = topology['topology'].hs["hub1"]
+
+    add_attributetype(supplier, 1, 'testAttribute', xorigin)
+
+    # Search for attribute with JSON option
+    schema = Schema(supplier)
+    attr_result = schema.query_attributetype('testAttribute', json=True)
+    # Verify the x-origin value is correct
+    assert attr_result['at']['x_origin'][0] == "user defined"
+
+    trigger_schema_push(topology, user.rdn, 1)
+    time.sleep(5)
```

Comment:
I am always wary about time.sleep in CI test (I have noticed that sometime the VM are frozen for up to 20 seconds).  IMHO in this case you should wait for replication between supplier and consumer